ubi_manifest/worker/tasks/depsolver/rpm_depsolver.py

```diff
@@ -152,6 +162,26 @@ def extract_and_resolve(self, content: set[UbiUnit]) -> None:
                     solved.add(req)
             self._unsolved -= solved
 
+    def resolve_files(
+        self, repos: list[YumRepository], blacklist: list[PackageToExclude]
+    ):
+        """
+        Get RPMs containing required files
+        """
+        content = set(self._file_req_rpms)
+        if self._required_files:
+            crit = create_or_criteria(["files"], [(f,) for f in self._required_files])
+            for rpm in f_proxy(
+                self._executor.submit(search_rpms, crit, repos, BATCH_SIZE_RPM)
+            ):
+                for filename in rpm.files:
+                    if filename in self._required_files:
+                        content.add(rpm)
+                        self._required_files.remove(filename)
```

Comment:
I think it's wrong to resolve anything with this set of rpms (it's a set all rpms that are returned from query and they are not filtered to their latest version).

I would do the same as in `self._what_provides()` - just query and return the latest rpms.

Reply:
this method does return `get_n_latest_from_content` but I guess I can remove the bit in the middle verifying the search result content against `_required_files` if that doesn't seem necessary to you